android_jni/avifandroidjni/src/main/java/com/avif/android/AvifDecoder.java

```diff
@@ -0,0 +1,58 @@
+package com.avif.android;
+
+import android.graphics.Bitmap;
+import java.nio.ByteBuffer;
+
+/** An AVIF Decoder. AVIF Specification: https://aomediacodec.github.io/av1-avif/. */
+@SuppressWarnings("CatchAndPrintStackTrace")
+public class AvifDecoder {
+  static {
+    try {
+      System.loadLibrary("avif_android");
+    } catch (UnsatisfiedLinkError exception) {
+      exception.printStackTrace();
+    }
+  }
+
+  // This is a utility class and cannot be instantiated.
+  private AvifDecoder() {}
+
+  /** Contains information about the AVIF Image dimensions. */
+  public static class Info {
+    public int width;
+    public int height;
+  }
+
+  /**
+   * Returns true if the bytes in the buffer seem like an AVIF image.
+   *
+   * @param buffer The encoded image.
+   * @return true if the bytes seem like an AVIF image, false otherwise.
+   */
+  public static boolean isAvifStream(ByteBuffer buffer) {
+    return AvifDecoder.isAvifStream(buffer, buffer.remaining());
+  }
+
+  private static native boolean isAvifStream(ByteBuffer encoded, int length);
+
+  /**
+   * Parses the AVIF header and populates the Info.
+   *
+   * @param encoded The encoded AVIF image.
+   * @param length Length of the encoded buffer.
+   * @param info Output parameter whose fields will be populated.
+   * @return true on success and false on failure.
+   */
+  public static native boolean getInfo(ByteBuffer encoded, int length, Info info);
+
+  /**
+   * Decodes the AVIF image into the bitmap.
+   *
+   * @param encoded The encoded AVIF image.
+   * @param length Length of the encoded buffer.
+   * @param bitmap The decoded pixels will be copied into the bitmap.
+   * @return true on success and false on failure. A few possible reasons for failure are: 1) Input
+   *     was not valid AVIF. 2) Bitmap was not large enough to store the decoded image.
+   */
+  public static native boolean decode(ByteBuffer encoded, int length, Bitmap bitmap);
+}
```

Comment:
It would be good to add a test or an example program for the Java `AvifDecoder` class.

Reply:
That requires adding a lot more boiler plate and a sample app. I am not sure if we should carry a lot more boilerplate code and maintain it to demonstrate what is a fairly simple java class. So i will leave it out of this PR. :)